wifiphisher/interfaces.py

```diff
@@ -0,0 +1,440 @@
+    """
+    Exception class to raise in case of catching any errors in
+    set_interface_mode function.
+    """
+
+    def __init__(self, error_message):
+        Exception.__init__(self,
+                           ("An error has detected in set_interface_mode.\n" +
+                            error_message))
+
+
+class NoApInterfaceFound(Exception):
+    """
+    Exception class to raise in case of a not finding a valid AP interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "No interface with AP mode was found!")
+
+
+class NoMonitorInterfaceFound(Exception):
+    """
+    Exception class to raise in case of a not finding a valid monitor
+    interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "No interface with monitor mode was found!")
+
+
+class JammingInterfaceInvalid(Exception):
+    """
+    Exception class to raise in case of a invalid jamming interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "Invalid jamming interface (-jI) was given!")
+
+
+class ApInterfaceInvalid(Exception):
+    """
+    Exception class to raise in case of a invalid ap interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "Invalid AP interface (-aI) was given!")
+
+
+def check_compatibility(interface):
+    """
+    Check the compatibility of the interface in regards to both monitor
+    mode and AP mode.
+
+    Args:
+        interface (str): An interface to be checked.
+
+    Returns:
+        ((bool), (bool)): The first item indicates if the interface has monitor
+                          mode and the second item indicates if AP mode is
+                          available.
+
+    Dependencies:
+        word_in_sentence
+    """
+
+    # Initialize variable names
+    device_name = str()
+    has_monitor = bool()
+    has_ap = bool()
+
+    # run the command to get all the wireless devices
+    device_command = subprocess.Popen(["iw", "dev"], stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE)
+
+    # get a list of all the devices
+    devices = device_command.communicate()[0].split("\n")
+
+    # find the name of the device if available
+    for line in range(len(devices)):
+        if word_in_sentence(devices[line], r"\b{0}\b".format(interface)):
+            device_name = devices[line-1]
+
+    # in case the device doesn't exist return False
+    if not device_name:
+        return has_monitor, has_ap
+
+    # run the command to get all the available info for device
+    info_command = subprocess.Popen(["iw", device_name, "info"],
+                                    stdout=subprocess.PIPE, stderr=DEVNULL)
+
+    # list of all the info
+    info = info_command.communicate()[0].split("\n")
+
+    # check info to see if monitor or AP mode is supported
+    for line in info:
+        # set has_monitor to True if monitor mode is supported
+        if line == "\t\t * monitor":
```

Comment:
No, the idea of `constants.py` is to define all constants (not only those required by the main module). 

The reason is not only reusability. An advanced user that wants to configure the software should be able to open `constants.py` and mess with it, instead of trying to figure out which file to change. In this scenario, a user may be using a different version of `iw` tool on his box where the support of monitor mode is represented differently. He can now simply edit the `constants.py` file instead of opening/looking at other files.

Imagine `constants.py` as parameters for advanced use cases.